PROVESFlightControllerReference/Components/PacketProcessor/Authenticator.cpp

```
@@ -0,0 +1,127 @@
+    if (!dataBuffer || dataSize < Ccsds355_0_B_2_Cmac::kSecurityTrailerSize) {
+        return PacketAuthenticator::Result{PacketAuthenticator::Status::VerifyError, PSA_ERROR_INVALID_ARGUMENT};
+    }
+
+    // Initialize PSA crypto library (idempotent)
+    const psa_status_t initStatus = psa_crypto_init();
+    if (initStatus != PSA_SUCCESS) {
+        return PacketAuthenticator::Result{PacketAuthenticator::Status::InitError, initStatus};
+    }
+
+    // Parse the hex-encoded default key into raw bytes
+    uint8_t keyBytes[Ccsds355_0_B_2_Cmac::kSecurityTrailerSize];
```

Clang (14.0.6)

[warning] variable 'keyBytes' is not initialized

(cppcoreguidelines-init-variables)